runtime/near-vm-logic/src/dependencies.rs

```diff
@@ -32,40 +32,285 @@ pub trait MemoryLike {
+    ///
+    /// let mut external = MockedExternal::new();
+    ///
+    /// external.storage_set(b"key42", b"value1337").unwrap();
+    /// // Creates iterator and returns index
+    /// let index = external.storage_iter(b"key42").unwrap();
+    ///
+    /// assert_eq!(external.storage_iter_next(index), Ok(Some((b"key42".to_vec(), b"value1337".to_vec()))));
+    /// assert_eq!(external.storage_iter_next(index), Ok(None));
+    ///
+    /// external.storage_iter(b"not_existing_key").expect("should be ok");
+    /// ```
     fn storage_iter(&mut self, prefix: &[u8]) -> Result<IteratorIndex>;
 
+    /// Creates iterator in memory for a key range and returns its ID to use with `storage_iter_next`
+    ///
+    /// # Arguments
+    ///
+    /// * `start` - a start prefix in the storage to iterate on
+    /// * `end` - an end prefix in the storage to iterate on (exclusive)
+    ///
+    /// # Errors
+    ///
+    /// This function could return:
+    /// - HostErrorOrStorageError::StorageError on underlying DB failure
+    ///
+    ///
+    /// # Example
+    /// ```
+    /// use near_vm_logic::mocks::mock_external::MockedExternal;
+    /// use near_vm_logic::External;
+    ///
+    /// let mut external = MockedExternal::new();
+    ///
+    /// external.storage_set(b"key42", b"value1337").unwrap();
+    /// external.storage_set(b"key43", b"val").unwrap();
+    /// // Creates iterator and returns index
+    /// let index = external.storage_iter_range(b"key42", b"key43").unwrap();
+    ///
+    /// assert_eq!(external.storage_iter_next(index), Ok(Some((b"key42".to_vec(), b"value1337".to_vec()))));
+    /// // The second key is `key43`. Returns Ok(None), since the `end` parameter is exclusive
+    /// assert_eq!(external.storage_iter_next(index), Ok(None));
+    /// ```
     fn storage_iter_range(&mut self, start: &[u8], end: &[u8]) -> Result<IteratorIndex>;
 
+    /// Returns the current iterator value and advances the iterator
+    /// If there is no more values, returns Ok(None)
+    ///
+    /// See usage examples in `storage_iter` and `storage_iter_range`
+    ///
+    /// # Arguments
+    ///
+    /// * `iterator_idx` - an iterator ID, created by `storage_iter` or `storage_iter_range`
+    ///
+    /// # Errors
+    ///
+    /// This function could return:
+    /// - HostErrorOrStorageError::StorageError on underlying DB failure
+    /// ```
     fn storage_iter_next(
         &mut self,
         iterator_idx: IteratorIndex,
     ) -> Result<Option<(Vec<u8>, Vec<u8>)>>;
 
+    /// Removes iterator index added with `storage_iter` and `storage_iter_range`
     fn storage_iter_drop(&mut self, iterator_idx: IteratorIndex) -> Result<()>;
 
+    /// Creates a receipt which will be executed after `receipt_indices`
+    ///
+    /// # Arguments
+    ///
+    /// * `receipt_indices` - a list of receipt indices the new receipt is depend on
+    ///
+    /// # Example
+    /// ```
+    /// use near_vm_logic::mocks::mock_external::MockedExternal;
```

Comment:
I would hide the repeating parts like these from the docstring here and everywhere else.
```rust
use near_vm_logic::External;
let mut external = MockedExternal::new();
```
 https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example